### PR TITLE
Platform detect v2

### PR DESCRIPTION
**Pull request by mbelluzzo**
* Applied proposed changes
* Add patch to check platform name that should fix Coverity Issue: 122264

### CODE REVIEW

**Comment by anselmolsm** [comment]
LGTM

**Comment by mbelluzzo** [comment]
Merged first 2, going to refactor the last one to use a simple for instead an overkill regex as pointed by @barbieri

**Comment by mbelluzzo** [comment]
So coverity issue will live one more day ;-)

File: data/schemas/platform-detect.schema [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    "regex_array": {
+      "type": "array",
+      "minItems": 1,
+      "uniqueItems": true,
+      "items": { "type": "string" }
+    },
+
+    "rule": {
+      "type": "object",
+      "additionalProperties": false,
```
Comment (lpereira) [inline data/schemas/platform-detect.schema]: s/additionalProperties/additional_properties/

Comment (mbelluzzo) [inline data/schemas/platform-detect.schema]: It follows http://json-schema.org/latest/json-schema-validation.html#anchor65. I didn't find "additional_properties" among json-schema docs.

Comment (lpereira) [inline data/schemas/platform-detect.schema]: Ah, my bad. I didn't see it was a `.schema` file; thought it was something you were defining.

Comment (mbelluzzo) [inline data/schemas/platform-detect.schema]: ;-)

**Pull request closed without merging**